go-controller/pkg/cluster/node.go

```diff
@@ -163,72 +156,37 @@ func (cluster *OvnClusterController) StartClusterNode(name string) error {
 		return err
 	}
 
-	type readyFunc func(string, string) (bool, error)
-	var readyFuncs []readyFunc
+	nodeAnnotator := kube.NewNodeAnnotator(cluster.Kube, node)
 
-	// get gateway annotations
-	gwAnnotations, postReady, err := cluster.initGateway(node.Name, subnet.String())
-	if err != nil {
-		return err
-	}
-	readyFuncs = append(readyFuncs, GatewayReady)
+	wg := &sync.WaitGroup{}
+	waitErrors := make(chan error)
 
-	// Get management port annotations
-	mgmtPortAnnotations, err := CreateManagementPort(node.Name, subnet, clusterSubnets)
-	if err != nil {
+	// Initialize gateway resources on the node
+	if err := cluster.initGateway(node.Name, subnet.String(), nodeAnnotator, wg, waitErrors); err != nil {
 		return err
 	}
-	readyFuncs = append(readyFuncs, ManagementPortReady)
-
-	// Combine mgmtPortAnnotations and gwAnnotations into nodeAnnotations
-	nodeAnnotations := make(map[string]interface{})
-	for k, v := range mgmtPortAnnotations {
-		nodeAnnotations[k] = v
-	}
-	for k, v := range gwAnnotations {
-		nodeAnnotations[k] = v
-	}
-
-	wg.Add(len(readyFuncs))
 
-	// Set node annotations
-	err = cluster.Kube.SetAnnotationsOnNode(node, nodeAnnotations)
-	if err != nil {
-		return fmt.Errorf("Failed to set node %s annotation: %v", node.Name, nodeAnnotations)
+	// Initialize management port resources on the node
+	if err := createManagementPort(node.Name, subnet, nodeAnnotator, wg, waitErrors); err != nil {
+		return err
 	}
 
-	portName := "k8s-" + node.Name
-
-	logrus.Infof("Waiting for GatewayReady and ManagementPortReady on node %s", node.Name)
-	// Wait for the portMac to be created
-	for _, f := range readyFuncs {
-		go func(rf readyFunc) {
-			defer wg.Done()
-			err := wait.PollImmediate(500*time.Millisecond, 300*time.Second, func() (bool, error) {
-				return rf(node.Name, portName)
-			})
-			messages <- err
-		}(f)
+	// Set all the annotations on the node at one time
```

Comment:
This comment only makes sense (and is only relevant) if you know that it didn't used to work that way.